lib/ansiblelint/errors.py

```diff
@@ -19,19 +19,35 @@ def __repr__(self):
                                 self.filename, self.linenumber, self.line)
 
 
+class AnsibleLintInternalRule():
+    """Internal error likely caused by parsing failures."""
+
+    id = "999"
+    shortdesc = "Parsing Error"
+    description = "Ansible failed to file."
+
+
 class MatchError(ValueError):
     """Exception that would end-up as linter rule match."""
 
-    def __init__(self, message, rule=None):
+    def __init__(self, message, rule=AnsibleLintInternalRule, filename="", linenumber=0, line=None):
```

Comment:
Please don't use mutable defaults like a class or a string. Use `None` instead.

Reply:
Current implementation relies on `rule` being an instance of a rule and checks for specific attributes. We would require several changes in order to make it work without a rule.

I think we can keep the code complexity lower by assuring that a match always has a rule, even if is a placeholder one. Also better for type checking.

Reply:
Still, you should use a None-idiom instead. Mutables in func/meth args is straight bad.

Reply:
maybe this is an option:
```python
def __init__(self, message, rule=None, filename="", linenumber=0, line=None):
    # ...
    if rule is None:
        rule = AnsibleLintInternalRule
    # ...
```